### PR TITLE
Add 'use-429-headers-for-rate-limit' rule

### PR DESCRIPTION
**Pull request by s0k0**

### CODE REVIEW

File: src/main/java/de/zalando/zally/rules/Use429ForRateLimitRule.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            return violations;
+        }
+        for(Map.Entry<String, Path> path: swagger.getPaths().entrySet())
+            for(Map.Entry<HttpMethod, Operation> op : path.getValue().getOperationMap().entrySet())
+                for(Map.Entry<String, Response> response: op.getValue().getResponses().entrySet()) {
+                    if ("429".equals(response.getKey()) && (
+                            response.getValue().getHeaders() == null
+                            || !response.getValue().getHeaders().containsKey("Retry-After")
+                            || !response.getValue().getHeaders().keySet().containsAll(X_RATE_LIMIT_TRIO)
+                        )
```
Comment (mkulak) [inline src/main/java/de/zalando/zally/rules/Use429ForRateLimitRule.java]: Consider introducing local variable for `response.getValue().getHeaders()`

File: src/test/resources/api_spp.json [hunk 2]
```diff
@@ -125,7 +128,10 @@
             "description" : "Client requested a product incorrectly thinking it was a coffee pot"
           },
           "429" : {
-            "description" : "Too many requests - the caller has issued too many requests, and is being rate limited.\nPlease refer to https://pages.github.bus.zalan.do/FashionInsightsCentre/spp-documentation/spp-api/rate-limiting\n"
+            "description" : "Too many requests - the caller has issued too many requests, and is being rate limited.\nPlease refer to https://pages.github.bus.zalan.do/FashionInsightsCentre/spp-documentation/spp-api/rate-limiting\n",
+            "headers": {
+              "X-RateLimit-Limit": 100
+            }
```
Comment (mkulak) [inline src/test/resources/api_spp.json]: Please don't modify api_* files directly - these are real examples from real projects. Copy it into fixtures, give it appropriate name and modify according to expectations in your test

File: src/test/java/de/zalando/zally/rules/Use429ForRateLimitRuleTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+public class Use429ForRateLimitRuleTest {
+
+    @Test
+    public void responseUse429ForRateLimitRuleNoViolationsJson() throws IOException {
+        Use429ForRateLimitRule rule = new Use429ForRateLimitRule();
+        Swagger swagger = new SwaggerParser().read("/api_spp.json");
+        List<Violation> violations = rule.validate(swagger);
+        assertThat(violations.size()).isEqualTo(9);
+        assertTrue(violations.stream().allMatch(v -> ViolationType.MUST.equals(v.getViolationType())));
+        assertTrue(violations.stream().allMatch(v -> "Use 429 With Header For Rate Limits".equals(v.getTitle())));
```
Comment (mkulak) [inline src/test/java/de/zalando/zally/rules/Use429ForRateLimitRuleTest.java]: I would rather check for specific paths: see for example `AvoidJavascriptKeywordsRuleTest`
```java
        List<String> results = new AvoidJavascriptKeywordsRule()
                .validate(invalidSwagger)
                .stream()
                .map(v -> v.getPath().get())
                .collect(Collectors.toList());
        assertThat(results).hasSameElementsAs(Arrays.asList("Pet.return", "Pet.typeof"));
```

**Comment by s0k0** [comment]
@mkulak please review again

**Comment by mkulak** [comment]
👍

**Pull request merged**